### ISSUE TITLE
Smarty Compiler: Syntax error unknown tag

### ISSUE THREAD

**Issue by ILFront-End** [smarty-php/smarty#375]
https://stackoverflow.com/questions/44952488/smarty-compiler-syntax-error-unknown-tag

please tell me hwo to solve this problem,thanks!

**Comment by ophian** [comment]
Hi
Apart from making sense or not, it does work.

test,php
```Php
{function name=menu level=0}
  <ul class="level{$level}">
  {foreach $data as $entry}
    {if is_array($entry)}
      <li>{$entry@key} {menu data=$entry level=$level+1}</li>
    {else}
      <li>{$entry}</li>
    {/if}
  {/foreach}
  </ul>
{/function}

{extends file='./base.tpl'}

{block name=head}
  <link href="/css/mypage.css" rel="stylesheet" type="text/css"/>
  <script src="/js/mypage.js"></script>
{/block}

{block name=body}
    {$menu = ['item1','item2','item3' => ['item3-1','item3-2','item3-3' => ['item3-3-1','item3-3-2']],'item4']}
    {menu data=$menu}
{/block}
```
base.tpl
```Smarty
<html>
  <head>
   {block name="head"}Default Head{/block}
  </head>
  <body>
    <p> {$smarty.version} </p>
    <div>
      {block "body"}Default Body{/block}
    </div>
  </body>
</html>
```
Output:
```html
<html>
  <head>
    <link href="/css/mypage.css" rel="stylesheet" type="text/css"/>
    <script src="/js/mypage.js"></script>
  </head>
  <body>
  <p> 3.1.32-dev-11 </p>
  <div>
    <ul class="level0">
      <li>item1</li>
      <li>item2</li>
      <li>item3 
        <ul class="level1">
          <li>item3-1</li>
          <li>item3-2</li>
          <li>item3-3 
            <ul class="level2">
              <li>item3-3-1</li>
              <li>item3-3-2</li>
            </ul>
          </li>
        </ul>
      </li>
      <li>item4</li>
    </ul>
  </div>
</body>
</html>
```

Result:
3.1.32-dev-11

    item1
    item2
    item3
        item3-1
        item3-2
        item3-3
            item3-3-1
            item3-3-2
    item4

**Comment by uwetews** [comment]
This problem  has been solved with the 3.1.31 release
